src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticAlteration.java

```diff
@@ -239,8 +237,8 @@ public HashMap<Integer,HashMap<Integer, String>> getGeneticAlterationMapForEntit
                 int geneticEntityId = rs.getInt("GENETIC_ENTITY_ID");
                 String values = rs.getString("VALUES");
                 //hm.debug..
-                String valueParts[] = values.split(DELIM);
-                for (int i=0; i<valueParts.length; i++) {
+                String valueParts[] = values.split(DELIM, -1);
+                for (int i = 0; i < orderedSampleList.size(); i++) {
```

Comment:
the important change here is adding -1 as the second argument to the `split` function. We don't want to trim empty values at the end! Each position is important. I agree about throwing an exception.

Reply:
I remembered: I iterate over `orderedSampleList` instead of `valueParts` because the database has values recorded with the trailing delimiter. So 3 values would be recorded as the following: `1,2,3,`. I am afraid we can't clean it up that easily as it'd require migration to clean the data in the database.

Reply:
see here https://github.com/cBioPortal/cbioportal-core/pull/47/commits/37dcc20b88904e0737dedd14261a6b7e089ddc02